File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/basesection/activities/Splash.java
package com.baytrendy.shopifyapp.basesection.activities;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000L\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\u000e\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0010\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0000\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\u0010\u0010\u000f\u001a\u00020\u00102\u0006\u0010\u0011\u001a\u00020\u0006H\u0002J\u0010\u0010\u0012\u001a\u00020\u00102\u0006\u0010\u0013\u001a\u00020\u0014H\u0002J\u0010\u0010\u0012\u001a\u00020\u00102\u0006\u0010\u0013\u001a\u00020\u0015H\u0002J\u0012\u0010\u0016\u001a\u00020\u00102\b\u0010\u0017\u001a\u0004\u0018\u00010\u0018H\u0014J\u0010\u0010\u0019\u001a\u00020\u00102\u0006\u0010\u001a\u001a\u00020\u001bH\u0002J\u0010\u0010\u0019\u001a\u00020\u00102\u0006\u0010\u001a\u001a\u00020\u001cH\u0002R\u0010\u0010\u0003\u001a\u0004\u0018\u00010\u0004X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0005\u001a\u0004\u0018\u00010\u0006X\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\u0007\u001a\u0004\u0018\u00010\bX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u001e\u0010\t\u001a\u00020\n8\u0006@\u0006X\u0087.\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u000b\u0010\f\"\u0004\b\r\u0010\u000e\u00a8\u0006\u001d"}, d2 = {"Lcom/shopify/shopifyapp/basesection/activities/Splash;", "Landroidx/appcompat/app/AppCompatActivity;", "()V", "binding", "Lcom/shopify/shopifyapp/base/databinding/MSplashBinding;", "product_id", "", "splashmodel", "Lcom/shopify/shopifyapp/basesection/viewmodels/SplashViewModel;", "viewModelFactory", "Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "getViewModelFactory", "()Lcom/shopify/shopifyapp/utils/ViewModelFactory;", "setViewModelFactory", "(Lcom/shopify/shopifyapp/utils/ViewModelFactory;)V", "consumeErrorResponse", "", "error", "consumeResponse", "reponse", "Lcom/shopify/shopifyapp/utils/FireBaseResponse;", "Lcom/shopify/shopifyapp/utils/LocalDbResponse;", "onCreate", "savedInstanceState", "Landroid/os/Bundle;", "renderSuccessResponse", "data", "Lcom/google/firebase/database/DataSnapshot;", "Lcom/shopify/shopifyapp/dbconnection/entities/AppLocalData;", "base_debug"})
public final class Splash extends androidx.appcompat.app.AppCompatActivity {
    @org.jetbrains.annotations.NotNull()
    @javax.inject.Inject()
    public com.baytrendy.shopifyapp.utils.ViewModelFactory viewModelFactory;
    private com.baytrendy.shopifyapp.basesection.viewmodels.SplashViewModel splashmodel;
    private com.baytrendy.shopifyapp.databinding.MSplashBinding binding;
    private java.lang.String product_id;
    private java.util.HashMap _$_findViewCache;
    
    @org.jetbrains.annotations.NotNull()
    public final com.baytrendy.shopifyapp.utils.ViewModelFactory getViewModelFactory() {
        return null;
    }
    
    public final void setViewModelFactory(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.utils.ViewModelFactory p0) {
    }
    
    @java.lang.Override()
    protected void onCreate(@org.jetbrains.annotations.Nullable()
    android.os.Bundle savedInstanceState) {
    }
    
    private final void consumeErrorResponse(java.lang.String error) {
    }
    
    private final void consumeResponse(com.baytrendy.shopifyapp.utils.LocalDbResponse reponse) {
    }
    
    private final void consumeResponse(com.baytrendy.shopifyapp.utils.FireBaseResponse reponse) {
    }
    
    private final void renderSuccessResponse(com.baytrendy.shopifyapp.dbconnection.entities.AppLocalData data) {
    }
    
    private final void renderSuccessResponse(com.google.firebase.database.DataSnapshot data) {
    }
    
    public Splash() {
        super();
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/basesection/activities/Splash.kt
package com.baytrendy.shopifyapp.basesection.activities

import android.annotation.SuppressLint
import android.app.job.JobInfo
import android.app.job.JobScheduler
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.PersistableBundle
import android.provider.Settings
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.ktx.auth
import com.google.firebase.database.DataSnapshot
import com.google.firebase.ktx.Firebase
import com.shopify.buy3.Storefront
import com.baytrendy.shopifyapp.MyApplication
import com.baytrendy.shopifyapp.MyApplication.Companion.context
import com.baytrendy.shopifyapp.MyApplication.Companion.dataBaseReference
import com.baytrendy.shopifyapp.MyApplication.Companion.firebaseapp
import com.baytrendy.shopifyapp.MyApplication.Companion.getmFirebaseSecondanyInstance
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.databinding.MSplashBinding
import com.baytrendy.shopifyapp.basesection.models.CommanModel
import com.baytrendy.shopifyapp.basesection.viewmodels.SplashViewModel
import com.baytrendy.shopifyapp.basesection.viewmodels.SplashViewModel.Companion.featuresModel
import com.baytrendy.shopifyapp.dbconnection.entities.AppLocalData
import com.baytrendy.shopifyapp.homesection.activities.HomePage
import com.baytrendy.shopifyapp.productsection.activities.ProductView
import com.baytrendy.shopifyapp.trialsection.activities.TrialExpired
import com.baytrendy.shopifyapp.utils.*
import javax.inject.Inject

class Splash : AppCompatActivity() {
    @Inject
    lateinit var viewModelFactory: ViewModelFactory
    private var splashmodel: SplashViewModel? = null
    private var binding: MSplashBinding? = null
    private var product_id: String? = null
    private var tm: JobScheduler? = null
    private lateinit var auth: FirebaseAuth
    private var productTitle: String? = null
    private var productData: Storefront.Product? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.m_splash, null, false)
        setContentView(binding!!.root)
        (application as MyApplication).mageNativeAppComponent!!.doSplashInjection(this)
        splashmodel = ViewModelProvider(this, viewModelFactory).get(SplashViewModel::class.java)
        splashmodel!!.message.observe(this, Observer<String> { this.showToast(it) })
        splashmodel?.setPresentmentCurrencyForModel()
        splashmodel!!.filteredproducts!!.observe(this, Observer<MutableList<Storefront.ProductEdge>> { this.setProductData(it) })
        initializeFirebase();
        val mServiceComponent = ComponentName(this, com.baytrendy.shopifyapp.jobservicessection.JobScheduler::class.java)
        val builder = JobInfo.Builder(101, mServiceComponent)
        builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY)
        builder.setRequiresDeviceIdle(false)
        builder.setMinimumLatency(0)
        builder.setRequiresCharging(false)
        builder.setPersisted(true)
        val extras = PersistableBundle()
        extras.putLong("MageNativeTEST", 10000)
        builder.setExtras(extras)
        tm = getSystemService(Context.JOB_SCHEDULER_SERVICE) as JobScheduler
        tm?.schedule(builder.build())
        if (intent != null) {
            val appLinkIntent = intent
            val appLinkAction = appLinkIntent.action
            if (appLinkIntent.data != null) {
                if (appLinkIntent.data!!.getQueryParameters("pid") != null) {
                    try {
                        product_id = appLinkIntent.data!!.getQueryParameters("pid")[0]
                    } catch (e: Exception) {
                        e.printStackTrace()
                        var link_array = appLinkIntent.data!!.toString().replace("https://", "").split("/")
                        productTitle = link_array.get(link_array.size - 1)
                        splashmodel?.getProductsByKeywords(productTitle!!)
                    }
                }
            }
        }
    }

    private fun setProductData(product: MutableList<Storefront.ProductEdge>?) {
        productData = product?.get(0)?.node
    }

    private fun showToast(it: String?) {
        Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
    }

    private fun initializeFirebase() {
        auth = Firebase.auth(firebaseapp!!)
        auth.signInWithEmailAndPassword("sudhanshshah@magenative.com", "asdcxzasd")
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {

                        val user = auth.currentUser

                        dataBaseReference = getmFirebaseSecondanyInstance().getReference(Urls(context).shopdomain.replace(".myshopify.com", ""))
                        if (splashmodel!!.isLogin) {
                            splashmodel!!.refreshTokenIfRequired()
                        }
                        splashmodel!!.firebaseResponse().observe(this, Observer<FireBaseResponse> { this.consumeResponse(it) })
                        splashmodel!!.Response(resources.getString(R.string.shopdomain)).observe(this, Observer<LocalDbResponse> { this.consumeResponse(it) })
                        splashmodel!!.errorMessageResponse.observe(this, Observer<String> { this.consumeErrorResponse(it) })
                        splashmodel!!.getNotificationCompaign().observe(this, Observer { this.cartNotification(it) })
                        @SuppressLint("HardwareIds") val deviceId = Settings.Secure.getString(this.contentResolver, Settings.Secure.ANDROID_ID)
                        splashmodel!!.sendTokenToServer(deviceId)

                        /* Toast.makeText(baseContext, "Authentication success",
                                 Toast.LENGTH_SHORT).show()*/
                    } else {

                        /*    Toast.makeText(baseContext, "Authentication failed.",
                                    Toast.LENGTH_SHORT).show()*/
                    }

                }
    }

    private fun cartNotification(it: Boolean?) {
        if (featuresModel.abandoned_cart_compaigns) {
            tm?.cancel(101)
        }
    }

    private fun consumeErrorResponse(error: String) {
        Toast.makeText(this, error, Toast.LENGTH_LONG).show()
    }

    private fun consumeResponse(reponse: LocalDbResponse) {
        when (reponse.status) {
            Status.SUCCESS -> renderSuccessResponse(reponse.data!!)
            Status.ERROR -> {
                reponse.error!!.printStackTrace()
                consumeErrorResponse(resources.getString(R.string.errorString))
            }
        }
    }

    private fun consumeResponse(reponse: FireBaseResponse) {
        when (reponse.status) {
            Status.SUCCESS -> renderSuccessResponse(reponse.data!!)
            Status.ERROR -> {
                reponse.error!!.printStackTrace()
                consumeErrorResponse(resources.getString(R.string.errorString))
            }
        }
    }

    private fun renderSuccessResponse(data: AppLocalData) {
        val intent: Array<Intent?>
        if (!data.isIstrialexpire) {
            intent = arrayOfNulls(1)
            intent[0] = Intent(this@Splash, TrialExpired::class.java)
        } else {
            if (product_id != null) {
                intent = arrayOfNulls(2)
                val product = Intent(this@Splash, ProductView::class.java)
                product.putExtra("ID", product_id)
                intent[1] = product
            } else if (productData != null) {
                intent = arrayOfNulls(2)
                val product = Intent(this@Splash, ProductView::class.java)
                product.putExtra("product", productData)
                product.putExtra("tittle", productTitle)
                intent[1] = product
            } else {
                intent = arrayOfNulls(1)
            }
            val homepage = Intent(this@Splash, HomePage::class.java)
            intent[0] = homepage
        }
        Handler().postDelayed(Runnable {
            startActivities(intent)
            Constant.activityTransition(this)
            finish()
        }, 1000)
    }

    private fun renderSuccessResponse(data: DataSnapshot) {
        try {
            val value = data.getValue(String::class.java)
            Log.i("MageNative", "launch_screen : " + value)
            val model = CommanModel()
            model.imageurl = value
            binding!!.commonmodel = model
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}

